pkg/controller/direct/compute/forwardingrule_controller.go

```diff
@@ -318,6 +318,11 @@ func (a *forwardingRuleAdapter) Create(ctx context.Context, createOp *directbase
 	var err error
 	op := &gcp.Operation{}
 	if a.id.location == "global" {
+		// todo(yuhou): TF does not support networkTier field for global forwarding rule
+		// It will always use GCP's default value, which is "PREMIUM." Any value set by the user will be ignored and not sent to GCP.
+		// To align with the TF controller, I remove this field.
+		// Ideally, direct controller should support this field and validate that the value.
+		forwardingRule.NetworkTier = nil
```

Comment:
I don't quite get this (but I like this comment). Could you describe what the Compute GCP service does for this field? I guess I'm confused because from your description it looks like TF sets the default "PREMIUM", in which case this value would be the default "PREMIUM" as well? Or, is the "PREMIUM" set by the GCP service? If it is the latter, is it a no-op or returns errors?

Regarding the previous TF behavior, I guess it could relate to the spec auto-correct in `state-into-spec:merge` mode.

Reply:
TF controller for global forwarding rule does not have "networkTier" field(see their [documentation](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_global_forwarding_rule#argument-reference)), that means whatever value provided by user, will be ignored(I do no see this field in the request body). If the field is unspecified, GCP will default it to "PREMIUM", so I can see the response body has this field with "PREMIUM". I hope below example can explain that clearly:

Configuration yaml:
```
metadata:
 name: globalforwardingrule
spec:
 networkTier: "anything"
 (other fields)
```
request body with TF controller:
```
POST ...globalforwardingrule...
{
(other fields)
}
```
response body with TF controller:
```
GET ...gloablforwardingrule...
{
networkTier: "PREMIUM"
(other fields)
}
```
Quote the field description: `For regional ForwardingRule, the valid values are 'PREMIUM' and 'STANDARD'. For GlobalForwardingRule, the valid value is 'PREMIUM'. If this field is not specified, it is assumed to be 'PREMIUM'. ` I think that explains why TF does not support this field for global forwarding rule and uses GCP default value.

Reply:
Thanks for the explanation. I think 
1. we want to include the `networkTier` field in `status.observedState`. 
2. I think we don't need this line `forwardingRule.NetworkTier = nil`, with the previous TF-based `state-into-spec: merge` default, this guarantees backward compatibility for existing forwarding rules.